atlp-rwanda/e-commerce-lydia-32-bn
# feature: Setting up user roles and permissions

### What does this PR do?
This PR adds role and permission management to the system, allowing the admin to assign roles to users and manage permissions for each role.

### Description of Task to be completed?
- Create `Role`, `Permission`, and `RolePermission` models.
- Update the `User` model to include a `roleId` field.
- Implement functionality to add and remove permissions from roles.
- Document API endpoints with Swagger.

### How should this be manually tested?
1. Run migrations to create new tables.
2. Use the API endpoints to assign roles to users and manage permissions.
3. Verify changes in the database.

### Any background context you want to provide?
N/A

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### What are the relevant pivotal tracker stories?
#187570873
